raiden-dapp/src/components/dialogs/UdcDepositDialog.vue

```diff
@@ -135,7 +136,27 @@ export default class UdcDepositDialog extends Vue {
   }
 
   get valid(): boolean {
-    return /^[1-9]\d*$/.test(this.defaultUtilityTokenAmount);
+    let utilityTokenAmount: BigNumber;
+
+    try {
+      utilityTokenAmount = parseUnits(
+        this.defaultUtilityTokenAmount,
+        this.udcToken.decimals
+      );
+    } catch (err) {
+      utilityTokenAmount = Zero;
+    }
+
+    if (!this.mainnet) return utilityTokenAmount.gt(Zero);
+
+    const utilityTokenBalance = parseUnits(
+      this.utilityTokenBalance,
```

Comment:
Notice `this.utilityTokenBalance` is a getter which formats the already BigNumber `this.udcToken.balance`. It doesn't make sense to parse it back to BigNumber here. Use `this.udcToken.balance` directly.

Reply:
Ah, good point.